framework/PostProcessors/FTgate.py

```diff
@@ -0,0 +1,293 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Created on April 30, 2018
+
+@author: mandd
+"""
+
+from __future__ import division, print_function , unicode_literals, absolute_import
+import warnings
+warnings.simplefilter('default', DeprecationWarning)
+
+#Internal Modules---------------------------------------------------------------
+import MessageHandler
+from utils import utils
+from utils import xmlUtils as xmlU
+#Internal Modules End-----------------------------------------------------------
+
+#External Modules---------------------------------------------------------------
+import numpy as np
+import xml.etree.ElementTree as ET
+import copy
+import itertools
+from collections import OrderedDict
+#External Modules End-----------------------------------------------------------
+
+class FTgate():
+  """
+    This is the base class for all possible Boolean logic gates to be employed in Fault-Trees
+  """
+  def __init__(self, xmlNode):
+    """
+      Method that initializes the gate
+      @ In, xmlNode, xml.etree.Element, Xml element node
+      @ Out, None
+    """
+    self.name         = None
+    self.gate         = None
+    self.arguments    = []
+    self.negations    = []
+    self.params       = {}
+    self.allowedGates = {'not':1,'and':'inf','or':'inf','xor':'inf','iff':2,'nand':'inf','nor':'inf','atleast':'inf','cardinality':'inf','imply':2}
+
+    self.name = xmlNode.get('name')
+
+    for child in xmlNode:
+      if child.attrib:
+        self.params = child.attrib
+      if child.tag in self.allowedGates.keys():
+        self.gate = child.tag
+      else:
+        raise IOError('FTImporterPostProcessor Post-Processor; gate ' + str(child.tag) + ' : is not recognized. Allowed gates are: '+ str(self.allowedGates.keys()))
+
+    for node in findAllRecursive(xmlNode, 'gate'):
+      self.arguments.append(node.get('name'))
+
+    for node in findAllRecursive(xmlNode, 'basic-event'):
+      self.arguments.append(node.get('name'))
+
+    for node in findAllRecursive(xmlNode, 'house-event'):
+      self.arguments.append(node.get('name'))
+
+    for child in xmlNode:
+      for childChild in child:
+        if childChild.tag == 'not':
+          event = list(childChild.iter())
+          if len(event)>2:
+            raise IOError('FTImporterPostProcessor Post-Processor; gate ' + str(self.name) + ' contains a negations of multiple basic events')
```

Comment:
no message handler?

Reply:
This class is used by FTstructure class, FTstructure class is used by FTmodel external model plugin, currently, we can not pass the messageHandler to external model plugin, that's why the messageHandler is not available for this class.